lib/ansible/module_utils/oneview.py

```diff
@@ -0,0 +1,782 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (2016-2017) Hewlett Packard Enterprise Development LP
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import,
+                        division,
+                        print_function,
+                        unicode_literals)
+
+try:
+    from future import standard_library
+    HAS_FUTURE = True
+except:
+    HAS_FUTURE = False
+    pass
+import json
+import logging
+import os
+import collections
+from ansible.module_utils.basic import AnsibleModule
+from copy import deepcopy
+
+if HAS_FUTURE:
+    standard_library.install_aliases()
+
+logger = logging.getLogger(__name__)
+
+try:
+    from hpOneView.oneview_client import OneViewClient
+    from hpOneView.exceptions import (HPOneViewException,
+                                      HPOneViewTaskError,
+                                      HPOneViewValueError,
+                                      HPOneViewResourceNotFound)
+
+    HAS_HPE_ONEVIEW = True
+except ImportError:
+    HAS_HPE_ONEVIEW = False
+
+
+class OneViewModuleBase(object):
+    MSG_CREATED = 'Resource created successfully.'
+    MSG_UPDATED = 'Resource updated successfully.'
+    MSG_DELETED = 'Resource deleted successfully.'
+    MSG_ALREADY_PRESENT = 'Resource is already present.'
+    MSG_ALREADY_ABSENT = 'Resource is already absent.'
+    HPE_ONEVIEW_SDK_REQUIRED = 'HPE OneView Python SDK is required for this module.'
+    FUTURE_PACKAGE_REQUIRED = 'The Future Python package is required for this module.'
+
+    ONEVIEW_COMMON_ARGS = dict(
+        config=dict(required=False, type='str')
+    )
+
+    ONEVIEW_VALIDATE_ETAG_ARGS = dict(
+        validate_etag=dict(
+            required=False,
+            type='bool',
+            default=True)
+    )
+
+    resource_client = None
+
+    def __init__(self, additional_arg_spec=None, validate_etag_support=False):
+        """
+        OneViewModuleBase constructor.
+
+        Args:
+            additional_arg_spec (dict): Additional argument spec definition.
+            validate_etag_support (bool): Enables support to eTag validation.
+        """
+
+        argument_spec = self.__build_argument_spec(additional_arg_spec, validate_etag_support)
```

Comment:
Note: We never use the double underscore private convention.  We just use the single underscore private convention.